universal-application-tool-0.0.1/app/filters/LoggingFilter.java

```diff
@@ -37,6 +45,31 @@ public EssentialAction apply(EssentialAction next) {
                     long time = clock.millis() - startTime;
                     log.info(
                         "{}\t{}\t{}ms\t{}", request.method(), request.uri(), time, result.status());
+                    StringBuilder requestCookies = new StringBuilder();
+                    Iterator<Http.Cookie> requestCookieIterator = request.cookies().iterator();
+                    while (requestCookieIterator.hasNext()) {
```

Comment:
suggestion (non-blocking): as an alternative, you can do:

```java
for (Http.Cookie cookie : request.cookies()) {
  // do things
}
```

Reply:
I can't, unfortunately, it is a `Cookies` object rather than a collection.

Reply:
I think since it extends `Iterable` this should be possible, but no worries if not! I just so rarely see iterators these days